File: src/backend/Bennetr.Lego.Api/Bennetr.Lego.Api/Models/LegoSet.cs
namespace Bennetr.Lego.Api.Models;

public class LegoSet
{
    public string Id { get; set; }

    public Group Group { get; set; }

    public DateTime Created { get; set; }

    public DateTime Updated { get; set; }

    public string SetNumber { get; set; }

    public string SetName { get; set; }

    public int ReleaseYear { get; set; }

    public Uri ImageUri { get; set; }

    public int TotalParts { get; set; }

    public int PresentParts { get; set; }

    public bool ForSale { get; set; }

    public bool Finished { get; set; }
}

File: src/frontend/src/domain/models/LegoSet.ts
import { Instance, SnapshotIn, SnapshotOut, types } from 'mobx-state-tree';
import { MSTDateTime } from '$/utils';

const LegoSet = types.model('LegoSet', {
  id: types.string,
  created: MSTDateTime,
  updated: MSTDateTime,
  setId: types.string,
  setName: types.string,
  releaseYear: types.integer,
  imageUri: types.string,
  totalParts: types.integer,
  presentParts: types.integer,
  forSale: types.boolean,
  finished: types.boolean
});

export default LegoSet;

export interface ILegoSet extends Instance<typeof LegoSet> {
}

export interface ILegoSetSnapshotIn extends SnapshotIn<typeof LegoSet> {
}

export interface ILegoSetSnapshotOut extends SnapshotOut<typeof LegoSet> {
}
